### PR TITLE
ModuleNotFoundError in  in_interactive_session

### LINKED ISSUE #24690: core\common.py:496: ModuleNotFoundError
In circumstances that I cannot reproduce in a small code example, I get the following error:

When looking at common.py, it seems that __main__ does indeed not exist. Does anybody know why and under what circumstances this is called?

```

_tcp\work\win-na-x64-release\py3\mre_venv3\lib\site-packages\pandas\core\base.py:61: in __str__

    return self.__unicode__()

_tcp\work\win-na-x64-release\py3\mre_venv3\lib\site-packages\pandas\core\frame.py:659: in __unicode__

    width, _ = console.get_console_size()

_tcp\work\win-na-x64-release\py3\mre_venv3\lib\site-packages\pandas\io\formats\console.py:68: in get_console_size

    if com.in_interactive_session():

_tcp\work\win-na-x64-release\py3\mre_venv3\lib\site-packages\pandas\core\common.py:503: in in_interactive_session

    return check_main()

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

 

    def check_main():

>       import __main__ as main

E       ModuleNotFoundError: No module named '__main__'

 

_tcp\work\win-na-x64-release\py3\mre_venv3\lib\site-packages\pandas\core\common.py:496: ModuleNotFoundError

```

### PR DESCRIPTION
**Pull request by dickreuter**
- [x] closes #24690
- [na] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Review by TomAugspurger — COMMENTED** [review]
I'm still iffy on this. It's not clear to me what pylint is doing, and I'd prefer not to add workarounds if it's something that should be fixed upstream.

File: doc/source/whatsnew/v0.24.0.rst [hunk 1]
```diff
@@ -1748,6 +1748,7 @@ I/O
 - Bug in :meth:`DataFrame.to_stata`, :class:`pandas.io.stata.StataWriter` and :class:`pandas.io.stata.StataWriter117` where a exception would leave a partially written and invalid dta file (:issue:`23573`)
 - Bug in :meth:`DataFrame.to_stata` and :class:`pandas.io.stata.StataWriter117` that produced invalid files when using strLs with non-ASCII characters (:issue:`23573`)
 - Bug in :class:`HDFStore` that caused it to raise ``ValueError`` when reading a Dataframe in Python 3 from fixed format written in Python 2 (:issue:`24510`)
+- Bug in :func:`in_interactive_session` that caused it to raise ``ModuleNotFoundError`` (:issue:`24690`)
```
[the remark this answers is not in the document]
Comment (jreback) [reply doc/source/whatsnew/v0.24.0.rst]: yeah I don't think we even need this note

**Review by jreback — CHANGES_REQUESTED** [review]
just remove the note

**Comment by dickreuter** [comment]
@rejack, remove the note

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Comment by dickreuter** [comment]
Not sure I understand. The pylint issue is affecting both py2 and py3 I think. Or why do you think there’s a difference?

**Comment by TomAugspurger** [comment]
From the linked bug report in
https://github.com/PyCQA/pylint/blob/master/pylint/lint.py#L153 to
https://bugs.python.org/issue10845,
it seems to be fixed in python 3.2. So it's possible that pylint could
apply that mock only for Python <=2.7

**Comment by dickreuter** [comment]
I see. I have submitted a PR for Pylint as well to remove the mock, but
their release cycle is very slow.

**Pull request merged**